c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -201,11 +209,12 @@ struct THCCachingAllocator
 
     size = round_size(size);
 
-    DeviceStats &stats = get_stats_for_device(device);
-
     Block search_key(device, stream, size);
     auto& pool = get_pool(size);
 
+    DeviceStats& stats = get_stats_for_device(device);
+    const std::vector<StatType> stat_types = {StatType::AGGREGATE, get_stat_type_for_pool(pool)};
```

Comment:
Hmm. Allocating a vector for every malloc seems unnecessary, especially since this is fixed size. Instead, you could make this be an initializer list (or a std::array) and then have the function that currently takes `const std::vector<StatType>&` take an `ArrayRef<StatType>` instead. (I don't know if this dynamic allocation matters or not, but it can be avoided so easily, so we should take the microoptimization.)

Reply:
Yep I had considered this but it felt like it would be a slight micro-optimization. The reason I biased against this was primarily if there would be multiple stat types in the future, and tensors would therefore have a varying number of stat types.

I suppose I could just make this a bitset, though – seems cleaner than a vector or array now that I think about it.